x-pack/solutions/security/plugins/security_solution/server/lib/telemetry/tasks/custom_response_actions_rule.ts

```diff
@@ -0,0 +1,142 @@
+    ) => {
+      const mdc = { task_id: taskId, task_execution_period: taskExecutionPeriod };
+      const log = newTelemetryLogger(logger.get('response_actions_rules'), mdc);
+      const usageCollector = sender.getTelemetryUsageCluster();
+      const usageLabelEndpointPrefix: string[] = [
+        'security_telemetry',
+        'endpoint-response-actions-rules',
+      ];
+      const usageLabelOsqueryPrefix: string[] = [
+        'security_telemetry',
+        'osquery-response-actions-rules',
+      ];
+      const trace = taskMetricsService.start(taskType);
+
+      log.l('Running response actions rules telemetry task');
+
+      try {
+        const [clusterInfoPromise, licenseInfoPromise] = await Promise.allSettled([
+          receiver.fetchClusterInfo(),
+          receiver.fetchLicenseInfo(),
+        ]);
+
+        const clusterInfo = safeValue(clusterInfoPromise);
+        const licenseInfo = safeValue(licenseInfoPromise);
+
+        const {
+          body: { aggregations },
+        } = (await receiver.fetchResponseActionsRules()) ?? {};
+
+        if (!aggregations || !aggregations.actionTypes) {
+          log.debug('no custom response action rules found');
+          await taskMetricsService.end(trace);
+          return 0;
+        }
+
+        const responseActionRules = (
+          aggregations as unknown as ResponseActionsRuleResponseAggregations
+        ).actionTypes.buckets.reduce<ResponseActionRules>((acc, agg) => {
+          if (agg.key === '.endpoint') {
+            acc.endpoint = agg.rulesInfo.buckets.map((rule) => rule.key);
+          } else if (agg.key === '.osquery') {
+            acc.osquery = agg.rulesInfo.buckets.map((rule) => rule.key);
+          }
+          return acc;
+        }, {} as ResponseActionRules);
+
+        const shouldNotProcessTelemetry =
+          responseActionRules.endpoint === undefined ||
+          responseActionRules.osquery === undefined ||
+          responseActionRules.endpoint.length === 0 ||
+          responseActionRules.osquery.length === 0;
+
+        if (shouldNotProcessTelemetry) {
+          return 0;
+        }
+
+        const responseActionsRulesTelemetryData = responseActionsCustomRuleTelemetryData(
+          responseActionRules,
+          clusterInfo,
+          licenseInfo
+        );
+
+        log.l('Custom response actions rules data', {
+          data: JSON.stringify(responseActionsRulesTelemetryData),
+        });
+
+        usageCollector?.incrementCounter({
+          counterName: createUsageCounterLabel(usageLabelEndpointPrefix),
+          counterType: 'response_actions_rules_count',
+          incrementBy: responseActionsRulesTelemetryData.response_actions.endpoint.count,
+        });
+
+        usageCollector?.incrementCounter({
+          counterName: createUsageCounterLabel(usageLabelOsqueryPrefix),
+          counterType: 'response_actions_rules_count',
+          incrementBy: responseActionsRulesTelemetryData.response_actions.osquery.count,
```

Comment:
Did you want both counters to have the same type? or change the `counterType` for `endpoint` and `osquery`?

Reply:
What are the published docs? I am using a query aggregation to keep a count of rules that have `osquery` or `endpoint` actions, and yes I do want to see the total count for each separately without one merging into another. I was testing this on the telemetry staging cluster and I am seeing some weird results, that might suggest that having a single counter type is adding up osquery and endpoint rule count into one instead of keeping them separate. 🤔 

I guess what I'm asking is does keeping two counter types ensure the counts are separate?